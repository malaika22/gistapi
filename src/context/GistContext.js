import { createContext, useEffect, useState } from 'react';
import { useFetch } from '../hooks/useFetch';
import { getPublicGists } from '../services/gistService';
import PropTypes from 'prop-types';

export const GistContext = createContext();

export const GistProvider = ({ children }) => {
  const { data, loading: fetchGistsLoading } = useFetch(getPublicGists);
  const [gistList, setGistList] = useState([]);

  useEffect(() => {
    setGistList(data);
  }, [data]);

  return (
    <GistContext.Provider value={{ gistList, setGistList, fetchGistsLoading }}>
      {children}
    </GistContext.Provider>
  );
};

GistProvider.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired
};
