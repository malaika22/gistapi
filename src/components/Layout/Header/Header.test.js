import { render } from '@testing-library/react';
import Header from './Header';
import { GistProvider } from 'context/GistContext';

describe('Header', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <GistProvider>
        <Header />
      </GistProvider>
    );
    expect(baseElement).toBeTruthy();
  });
  it('should match snapshot', () => {
    const { asFragment } = render(
      <GistProvider>
        <Header />
      </GistProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
