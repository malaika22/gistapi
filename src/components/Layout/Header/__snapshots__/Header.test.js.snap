// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-fFGjHI dISJgE"
  >
    <span
      class="mega-octicon octicon-mark-github"
    />
    <div
      class="sc-bgqQcB ighHLN"
    >
      <div
        class="sc-gTRrQi cLuoSN"
      >
        <span
          class="octicon octicon-search"
        />
        <input
          class="sc-ewnqHT eQrKCi"
          placeholder="Search Gists for the username"
          value=""
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
